weave/reference/python-sdk.mdx

```diff
@@ -8,275 +8,1575 @@ description: "Python SDK reference for weave"
+- `description`: `str | None`
+- `ref`: `trace.refs.ObjectRef | None`
+- `history`: `list[typing.Any]`
+
+---
+
+<a href="https://github.com/wandb/weave/blob/0.52.20/weave/trace_server/interface/builtin_object_classes/annotation_spec.py#L12"><img align="right" src="https://img.shields.io/badge/-source-cccccc?style=flat-square" /></a>
+
+## <kbd>class</kbd> `AnnotationSpec`
+
+**Pydantic Fields:**
+
+- `name`: `str | None`
+- `description`: `str | None`
+- `field_schema`: `dict[str, typing.Any]`
+- `unique_among_creators`: `<class 'bool'>`
+- `op_scope`: `list[str] | None`
+
+<a href="https://github.com/wandb/weave/blob/0.52.20/weave/trace_server/interface/builtin_object_classes/annotation_spec.py#L47"><img align="right" src="https://img.shields.io/badge/-source-cccccc?style=flat-square" /></a>
+
+### <kbd>classmethod</kbd> `preprocess_field_schema`
+
+```python
+preprocess_field_schema(data: dict[str, Any]) → dict[str, Any]
+```
+
+---
+
+<a href="https://github.com/wandb/weave/blob/0.52.20/weave/trace_server/interface/builtin_object_classes/annotation_spec.py#L92"><img align="right" src="https://img.shields.io/badge/-source-cccccc?style=flat-square" /></a>
+
+### <kbd>classmethod</kbd> `validate_field_schema`
+
+```python
+validate_field_schema(schema: dict[str, Any]) → dict[str, Any]
+```
+
+---
+
+<a href="https://github.com/wandb/weave/blob/0.52.20/weave/trace_server/interface/builtin_object_classes/annotation_spec.py#L103"><img align="right" src="https://img.shields.io/badge/-source-cccccc?style=flat-square" /></a>
+
+### <kbd>method</kbd> `value_is_valid`
+
+```python
+value_is_valid(payload: Any) → bool
+```
+
+Validates a payload against this annotation spec's schema. 
+
+**Args:**
+ 
+
+ - <b>`payload`</b>: The data to validate against the schema
+**Returns:**
+ 
+ - <b>`bool`</b>:  True if validation succeeds, False otherwise 
+
+---
+
+<a href="https://github.com/wandb/weave/blob/0.52.20/weave/type_handlers/Audio/audio.py#L81"><img align="right" src="https://img.shields.io/badge/-source-cccccc?style=flat-square" /></a>
+
+## <kbd>class</kbd> `Audio`
+A class representing audio data in a supported format (wav or mp3). 
+
+This class handles audio data storage and provides methods for loading from different sources and exporting to files. 
+
+**Attributes:**
+ 
+ - <b>`format`</b>:  The audio format (currently supports 'wav' or 'mp3') 
+ - <b>`data`</b>:  The raw audio data as bytes 
+
+**Args:**
+ 
+
+ - <b>`data`</b>: The audio data (bytes or base64 encoded string)
+ - <b>`format`</b>: The audio format ('wav' or 'mp3')
+ - <b>`validate_base64`</b>: Whether to attempt base64 decoding of the input data
+**Raises:**
+ 
+ - <b>`ValueError`</b>:  If audio data is empty or format is not supported 
+
+<a href="https://github.com/wandb/weave/blob/0.52.20/weave/type_handlers/Audio/audio.py#L106"><img align="right" src="https://img.shields.io/badge/-source-cccccc?style=flat-square" /></a>
+
+### <kbd>method</kbd> `__init__`
+
+```python
```

Comment:
These lines validate the Lazydocs workaround for the bug https://github.com/ml-tooling/lazydocs/issues/97.

The diff of this file validates that the landing page is updated correctly rather than being cleared out.